### PR TITLE
Add 11 AS3 classes

### PR DESCRIPTION
**Pull request by huwdp**
**This commit adds:**

- flash.accessibility.Accessibility
- flash.globalization.CurrencyParseResult
- flash.media.AVNetworkingParams
- flash.media.AVTagData
- flash.media.VideoStreamSettings
- flash.net.NetStreamPlayOptions
- flash.net.URLRequestDefaults
- flash.system.IME
- flash.system.JPEGLoaderContext
- flash.text.TextRenderer
- flash.ui.MouseCursorData

These classes I have identified as ones that can be implemented/stubbed. I'm happy to remove any file(s) not needed :). Code generated by [ActionScript_Event_Builder project ](https://github.com/golfinq/ActionScript_Event_Builder) and modified by me slightly.

**To-Do:**
- [x] Check code and code quality
- [x] Test flash.accessibility.Accessibility
- [x] Test flash.globalization.CurrencyParseResult
- [x] Test flash.media.AVNetworkingParams
- [x] Test flash.media.AVTagData
- [ ] ~~Test flash.media.VideoStreamSettings~~
- [x] Test flash.net.NetStreamPlayOptions
- [ ] ~~Test flash.net.URLRequestDefaults~~
- [ ] ~~Test flash.system.IME~~ (all stubs)
- [x] Test flash.system.JPEGLoaderContext
- [ ] ~~Test flash.text.TextRenderer~~ (all stubs)
- [ ] ~~Test flash.ui.MouseCursorData~~ (all stubs)

### CODE REVIEW

File: core/src/avm2/globals/flash/ui/MouseCursorData.as [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+// It won't be regenerated in the future, so feel free to edit and/or fix
+
+package flash.ui
+{
+    import flash.geom.Point;
+    
+    public final class MouseCursorData
+    {
+        // A Vector of BitmapData objects containing the cursor image or images.
+        public var data: Vector;
```
Comment (Lord-McSweeney) [inline core/src/avm2/globals/flash/ui/MouseCursorData.as:13]: I think this should be `Vector.<BitmapData>`.

**Comment by Lord-McSweeney** [comment]
There are a number of empty constructors (such as:
```
        public function MicrophoneEnhancedOptions()
        {

        }
```
) that can be removed.

File: core/src/avm2/globals/flash/ui/MouseCursorData.as (hunk 1 above)
Comment (huwdp) [reply core/src/avm2/globals/flash/ui/MouseCursorData.as]: Thank you!

**Comment by huwdp** [comment]
> There are a number of empty constructors (such as:
> 
> ```
>         public function MicrophoneEnhancedOptions()
>         {
> 
>         }
> ```
> 
> ) that can be removed.

This is done

File: core/src/avm2/globals/flash/accessibility/Accessibility.as [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+{
+    public final class Accessibility
+    {
+        // Indicates whether a screen reader is active and the application is communicating with it.
+        private var _active: Boolean;
+
+        // Tells Flash Player to apply any accessibility changes made by using the DisplayObject.accessibilityProperties property.
+        public static function updateProperties():void
+        {
+            stub_method("flash.accessibility.Accessibility", "updateProperties");
```
Comment (Lord-McSweeney) [inline core/src/avm2/globals/flash/accessibility/Accessibility.as:16]: I think you need an import for this stub (`import __ruffle__.stub_method`)

File: core/src/avm2/globals/flash/media/MicrophoneEnhancedOptions.as [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        public var echoPath: int;
+
+        // Indicates whether the Microphone input detected a voice.
+        public var isVoiceDetected: int;
+
+        // Controls enhanced microphone mode.
+        public var mode: String;
+
+        // Enable non-linear processing.
+        public var nonLinearProcessing: Boolean;
```
Comment (Lord-McSweeney) [inline core/src/avm2/globals/flash/media/MicrophoneEnhancedOptions.as:20]: This should default to `true`.

File: core/src/avm2/globals/flash/media/MicrophoneEnhancedOptions.as [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+// https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/media/MicrophoneEnhancedOptions.html
+// by https://github.com/golfinq/ActionScript_Event_Builder
+// It won't be regenerated in the future, so feel free to edit and/or fix
+
+package flash.media
+{
+    public final class MicrophoneEnhancedOptions
+    {
+        // Specifies the echo path (in milliseconds) used for acoustic echo cancellation.
+        public var echoPath: int;
```
Comment (Lord-McSweeney) [inline core/src/avm2/globals/flash/media/MicrophoneEnhancedOptions.as:11]: This should default to 128.

File: core/src/avm2/globals/flash/media/MicrophoneEnhancedOptions.as [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+        // Indicates whether the Microphone input detected a voice.
+        public var isVoiceDetected: int;
+
+        // Controls enhanced microphone mode.
+        public var mode: String;
```
Comment (Lord-McSweeney) [inline core/src/avm2/globals/flash/media/MicrophoneEnhancedOptions.as:17]: From the Flash docs:
Controls enhanced microphone mode. The default value is `MicrophoneEnhancedMode.FULL_DUPLEX` for all microphones that aren't USB. The default value for USB microphones is `MicrophoneEnhancedMode.HALF_DUPLEX`. See MicrophoneEnhancedMode for possible values and descriptions.

**Comment by Lord-McSweeney** [comment]
The AS3 docs say that most of these can be only assigned specific values- I think using a getter/setter for these might make more sense.

File: core/src/avm2/globals/flash/media/MicrophoneEnhancedOptions.as (hunk 5 above)
Comment (huwdp) [reply core/src/avm2/globals/flash/media/MicrophoneEnhancedOptions.as]: I've set this to the default option and added a stub. Please check if I'm doing this one correct.

**Comment by huwdp** [comment]
> The AS3 docs say that most of these can be only assigned specific values- I think using a getter/setter for these might make more sense.

This will be my next thing

**Comment by huwdp** [comment]
This is now ready! I could not create tests for VideoStreamSettings, URLRequestDefaults & IME.

**Review by Dinnerbone — APPROVED** [review]

**Comment by Dinnerbone** [comment]
Can you please rebase onto master? Thank you!

**Comment by huwdp** [comment]
Somehow a test for VideoStreamSettings ended up there, so I've just removed that

**Comment by Dinnerbone** [comment]
Thank you very much!

**Pull request merged**